electrumpersonalserver/server/peertopeer.py

```diff
@@ -0,0 +1,410 @@
+                self.sock.connect(self.remote_hostport)
+                self.sock.sendall(data)
+                break
+            except IOError as e:
+                if len(self.dns_seeds) == 0:
+                    raise e
+                else:
+                    # cycle to the next dns seed
+                    time.sleep(0.5)
+                    log.debug('connection attempts = ' + str(self.connection_attempts))
+                    self.connection_attempts -= 1
+                    if self.connection_attempts == 0:
+                        raise e
+                    self.dns_index = (self.dns_index + 1) % len(self.dns_seeds)
+                    self.remote_hostport = (self.dns_seeds[self.dns_index],
+                                            self.remote_hostport[1])
+
+        log.info('connected')
+        self.sock.settimeout(self.heartbeat_interval)
+        self.closed = False
+        try:
+            recv_buffer = b''
+            payload_length = -1  # -1 means waiting for header
+            command = None
+            checksum = None
+            while not self.closed:
+                try:
+                    recv_data = self.sock.recv(4096)
+                    if not recv_data or len(recv_data) == 0:
+                        raise EOFError()
+                    recv_buffer += recv_data
+                    # this is O(N^2) scaling in time, another way would be to store in a list
+                    # and combine at the end with "".join()
+                    # but this isnt really timing critical so didnt optimize it
+
+                    data_remaining = True
+                    while data_remaining and not self.closed:
+                        if payload_length == -1 and len(recv_buffer) >= HEADER_LENGTH:
+                            net_magic, command, payload_length, checksum = unpack('<I12sI4s', recv_buffer[:HEADER_LENGTH])
+                            recv_buffer = recv_buffer[HEADER_LENGTH:]
+                            if net_magic != self.magic:
+                                log.error('wrong MAGIC: ' + hex(net_magic))
+                                self.sock.close()
+                                break
+                            command = command.strip(b'\0')
+                        else:
+
+                            if payload_length >= 0 and len(recv_buffer) >= payload_length:
+                                payload = recv_buffer[:payload_length]
+                                recv_buffer = recv_buffer[payload_length:]
+                                if btc.bin_dbl_sha256(payload)[:4] == checksum:
+                                    self.p2p_message_handler.handle_message(self, command,
+                                        payload_length, payload)
+                                else:
+                                    log.error('wrong checksum, dropping message, cmd=' + command + ' payloadlen=' + str(payload_length))
```

Comment:
This should be changed to `log.debug()` because we safely deal with this case and the error probably caused by things outside our control, so theres no point using warning.